### PR TITLE
Updated Content Extractor Module

### PR DESCRIPTION
**Pull request by bsuryadevara**
- Added Haystack Converter API to convert CSV, PDF, DOCX to text.

### CODE REVIEW

File: examples/llm/common/content_extractor_module.py [hunk 1]
```diff
@@ -30,62 +36,122 @@
...
-            return f.read()
-    except Exception as e:
-        logger.error(f"Error reading file {file.path}: {e}")
-        return None
-
 
-def read_pdf_file(file):
+    outgoing_edges = 1
+
+    def convert(self,
```
Comment (drobison00) [inline examples/llm/common/content_extractor_module.py:53]:
```suggestion
    def convert_csv_to_document(self,
```

File: examples/llm/common/content_extractor_module.py [hunk 2]
```diff
@@ -30,62 +36,122 @@
...
+                id_hash_keys: typing.Optional[list[str]] = None) -> list[Document]:
+        """
+        Load a CSV file and convert it to Documents.
+
+        Parameters
+        ----------
+        file_path:
+            Path to the CSV file you want to convert.
+        meta:
+            A dictionary of metadata key-value pairs that you want to append to the returned document. It's optional.
```
Comment (drobison00) [inline examples/llm/common/content_extractor_module.py]:
```suggestion
            Optional dictionary of metadata key-value pairs that you want to append to the returned document.
```

File: examples/llm/common/content_extractor_module.py [hunk 3]
```diff
@@ -30,62 +36,122 @@
...
+        encoding:
+            Specifies the file encoding. It's optional. The default value is `UTF-8`.
```
Comment (drobison00) [inline examples/llm/common/content_extractor_module.py]:
```suggestion
            Optional file encoding format, default: `UTF-8`.
```

File: examples/llm/common/content_extractor_module.py [hunk 4]
```diff
@@ -30,62 +36,122 @@
...
+        -------
+        list[haystack.Document]
+            List of documents, 1 document per line in the CSV.
+        """
+        if not isinstance(file_path, list):
+            file_path = [file_path]
+
+        docs: list[Document] = []
+        if meta is None:
+            text_column_name = "page_content"
```
Comment (drobison00) [inline examples/llm/common/content_extractor_module.py]:
```suggestion
            text_column_name = "content"
```

For a CSV file, 'page_content' doesn't make sense.

File: examples/llm/common/content_extractor_module.py [hunk 5]
```diff
@@ -30,62 +36,122 @@
...
+        else:
+            text_column_name = meta["csv"]["text_column_name"]
+
+        for path in file_path:
+            df = pd.read_csv(path, encoding=encoding)
+            if len(df.columns) == 0 or (text_column_name not in df.columns):
```
Comment (drobison00) [inline examples/llm/common/content_extractor_module.py:94]: Again, I'm not a fan of the 'page_content' default here, I don't think it makes sense unless we're doing something like RSS, which is explicitly pulling 'page_content'.

File: examples/llm/common/content_extractor_module.py (hunk 1 above)
Comment (bsuryadevara) [reply examples/llm/common/content_extractor_module.py:53]: `convert` function is an abstractmethod of  Haystack BaseConverter API

**Pull request merged**